### PR TITLE
Migrate legacy tables

### PR DESCRIPTION
**Pull request by DenisCarriere**
- [x] 69c3f2d57c9ee4931bbeec8e45aa9c6194d06075 update params for `calculate_return`
  - remove dependency of `converter_t` struct
- [x] 3519a0078cc092050a18a1dc2daff69044e41a50 update migration steps
  - new action `cleartables` 
  - erase `reserves` & `converters` (scoped & no-scoped)
- [x] b87d90585b0a5123ff6a869e40d97a744871a83b update `get_reserve` method
  - added new struct `reserve`
  - `get_reserve` outputs `BancorConveter::reserve`

```c++
struct reserve {
    name        contract;
    uint64_t    weight;
    asset       balance;
};
```
- [x] 67b77412e5e9112cdd393779128d41cde6284a8e remove dependency of `reserves` TABLE
- [x] 7848f54c75db82144126c7d2afd8be6a9fded6cd remove dependency of `converters` TABLE

### CODE REVIEW

File: contracts/eos/BancorConverter/BancorConverter.hpp [hunk 1]
```diff
@@ -408,7 +435,7 @@ class [[eosio::contract]] BancorConverter : public contract { /*! \endcond */
         static uint128_t _by_cnvrt( asset balance, symbol_code converter_currency_code ) {
            return ( uint128_t{ balance.symbol.code().raw() } << 64 ) | converter_currency_code.raw();
         }
-        constexpr static double MAX_RATIO = 1000000.0;
+        constexpr static double MAX_WEIGHT = 1000000.0;
```
Comment (yudilevi) [inline contracts/eos/BancorConverter/BancorConverter.hpp:438]: can simply use PPM_RESOLUTION for both here

File: contracts/eos/BancorConverter/src/convert.cpp [hunk 2]
```diff
@@ -74,22 +72,22 @@ std::tuple<asset, double> BancorConverter::calculate_return(extended_asset from_
     }
     else {
         if (!incoming_smart_token) { // Reserve --> Smart
```
Comment (yudilevi) [inline contracts/eos/BancorConverter/src/convert.cpp:74]: can basically completely remove support for swaps between reserve <> smart token.
smart tokens are acquired by adding liquidity.
also, "smart token" isn't used anymore - it's just token (no need to migrate for this :) but would be helpful to rename wherever possible that doesn't require migration)

File: contracts/eos/BancorConverter/BancorConverter.hpp (hunk 1 above)
Comment (DenisCarriere) [reply contracts/eos/BancorConverter/BancorConverter.hpp]: Renamed ✅ 

https://github.com/bancorprotocol/contracts_eos/pull/72/commits/06c03d391c401b3089b97b7031fbac62490a8c8b

File: contracts/eos/BancorConverter/src/convert.cpp (hunk 2 above)
Comment (DenisCarriere) [reply contracts/eos/BancorConverter/src/convert.cpp:74]: Ok, good to know, perhaps we can create a new PR for everything that's `smart token` removal

Does this mean that the convert method only handles `quick_conversion`?

Comment (yudilevi) [reply contracts/eos/BancorConverter/src/convert.cpp:74]: It's actually out of the scope of this migration so we'll handle that separately later.

**Review by yudilevi — APPROVED** [review]

**Pull request merged**